kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/api/common/FilterRefTest.java

```diff
@@ -12,13 +12,17 @@
 
 class FilterRefTest {
 
+    // we knowingly use equals across types because we want the property that specific LocalRef types are equal to any other LocalRef
+    // with the same group, kind and name.
+    @SuppressWarnings("java:S5845")
     @Test
-    void shouldEqualAKafkaKindedAnyRef() {
```

Comment:
KafkaKinded didn't make much sense to me.  I made me think about Strimzi's Kafka resource.